### PR TITLE
Implemented Singleton Use Case

### PR DESCRIPTION
**Pull request by JeffeyL**

### CODE REVIEW

File: src/businessrules/singleton/usecases/DeleteSingletonUseCase.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        this.vendorRepository = vr;
+        this.errorHandler = em;
+        this.shopRepository = sr;
+        this.singletonLoader = new SingletonLoader(singletonrep, em);
+        this.vendorLoader = new VendorLoader(vr, em);
+    }
+
+    @SuppressWarnings("DuplicatedCode")
+    @Override
+    public boolean deleteSingleton(String vendorToken, String id) {
```
Comment (stephlum) [inline src/businessrules/singleton/usecases/DeleteSingletonUseCase.java:40]: When you delete a singleton, should all food items that contain that singleton get deleted too?

**Review by stephlum — COMMENTED** [review]
Looks good for the most part. Please see comments

File: src/businessrules/loaders/SingletonLoader.java [hunk 2]
```diff
@@ -1,11 +1,68 @@
...
+        ArrayList<Integer> arr = new ArrayList<>();
+        for(Object value : array){
+            arr.add((int) value);
+        }
+        return arr;
+    }
+
+    public static List<Addon> getAddonListFromJSONArray(JSONArray array){
+        ArrayList<Addon> arr = new ArrayList<>();
+        // TODO: implement this method
```
[the remark this answers is not in the document]
Comment (akashilangovan) [reply src/businessrules/loaders/SingletonLoader.java:65]: ^good point

**Review by akashilangovan — APPROVED** [review]
Minor issues but can merge if needed

**Pull request merged**